thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2760,26 +2760,11 @@ class CSpill : implements IRowWriter, public CSimpleInterface
         ::Release(writer);
         writer = NULL;
         spillFileIO->flush();
-        mergeStats(stats, this);
-        spillFile->noteSize(getStatistic(StSizeSpillFile));
+        mergeRemappedStats(stats, spillFileIO, diskToSpillStatsMap);
+        stats.setStatistic(StNumSpills, 1);
```

Comment:
setStatistic doesn't sound right?
close(CRuntimeStatisticCollection &stats) will be called multiple times (twice in 1 execution, potentially more if in loop).